In [1]:
import numpy as np
import pandas as pd

In [2]:
#read data and show first 5 rows
path = open(r"C:\Users\elegb\Desktop\Hamoye.csv") #renamed the datset to hamoye
data = pd.read_csv(path)
data.head()

Area Code     Area  Item Code         Item  Element Code  \
0          4  Algeria       2501   Population           511   
1          4  Algeria       2501   Population          5301   
2          4  Algeria       2901  Grand Total           664   
3          4  Algeria       2901  Grand Total           674   
4          4  Algeria       2901  Grand Total           684   

                                  Element             Unit     Y2014  \
0           Total Population - Both sexes     1000 persons  38924.00   
1                Domestic supply quantity      1000 tonnes      0.00   
2           Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3  Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4      Fat supply quantity (g/capita/day)     g/capita/day     80.06   

      Y2015     Y2016     Y2017     Y2018  
0  39728.00  40551.00  41389.00  42228.00  
1      0.00      0.00      0.00      0.00  
2   3379.00   3372.00   3341.00   3322.00  
3     94.35     94.72     92.82     91.83  
4     79.36     77.40     80.19     77.28

In [3]:
#descriptive statistics
data.describe()

Area Code     Item Code  Element Code          Y2014          Y2015  \
count  60943.000000  60943.000000  60943.000000   59354.000000   59395.000000   
mean     134.265576   2687.176706   3814.856456     134.196282     135.235966   
std       72.605709    146.055739   2212.007033    1567.663696    1603.403984   
min        4.000000   2501.000000    511.000000   -1796.000000   -3161.000000   
25%       74.000000   2562.000000    684.000000       0.000000       0.000000   
50%      136.000000   2630.000000   5142.000000       0.090000       0.080000   
75%      195.000000   2775.000000   5511.000000       8.340000       8.460000   
max      276.000000   2961.000000   5911.000000  176405.000000  181137.000000   

               Y2016          Y2017          Y2018  
count   59408.000000   59437.000000   59507.000000  
mean      136.555222     140.917765     143.758381  
std      1640.007194    1671.862359    1710.782658  
min     -3225.000000   -1582.000000   -3396.000000  
25%         0.000000       0.000000       0.000000  
50%         0.080000       0.100000       0.070000  
75%         8.430000       9.000000       9.000000  
max    185960.000000  190873.000000  195875.000000

In [4]:
#checking for missing values
data.isnull().sum()

Area Code          0
Area               0
Item Code          0
Item               0
Element Code       0
Element            0
Unit               0
Y2014           1589
Y2015           1548
Y2016           1535
Y2017           1506
Y2018           1436
dtype: int64

Missing values across the years vary.

In [5]:
#how many rows and columns are in our dataset
data.shape

(60943, 12)

The dataset consist of 60943 rows and 12 columns 

# 1. What is the total sum of Animal fat produced in 2014 and 2017?

In [6]:
data.groupby('Item')['Y2014'].sum()

Item
Alcohol, Non-Food        2403.00
Alcoholic Beverages    102410.11
Animal Products         11935.65
Animal fats            209460.54
Apples and products      9499.23
                         ...    
Vegetables, Other      155038.96
Vegetal Products       107145.19
Wheat and products     232670.13
Wine                     4497.36
Yams                   200396.96
Name: Y2014, Length: 119, dtype: float64

The total sum of Animal Fats produced in year 2014 = 209460.54

# 2. What is the mean and standard deviation across the whole dataset for the year 2015 to 3 decimal places?

In [7]:
round(data['Y2015'].agg(['mean','std']),3)

mean     135.236
std     1603.404
Name: Y2015, dtype: float64

The mean and standard deviation for the year 2015 are 135.236 and 1603.404 respectively.

# 3. What is the total number and percentage of missing data in 2016 to 2 decimal places?

In [8]:
Total =data['Y2015'].isnull().sum()
percentage = Total/(data['Y2016'].count() + Total)
print(round(Total))
round((percentage*100),2)

1548


2.54

The total number of missing data in 2016 = 1548 and the percenage of missing value is 2.54%

# 4. Which year had the highest correlation with ‘Element Code’? 

In [9]:
data.corr()

Area Code  Item Code  Element Code     Y2014     Y2015  \
Area Code      1.000000  -0.005159     -0.000209  0.006164  0.005472   
Item Code     -0.005159   1.000000     -0.024683  0.021722  0.020857   
Element Code  -0.000209  -0.024683      1.000000  0.024457  0.023889   
Y2014          0.006164   0.021722      0.024457  1.000000  0.994647   
Y2015          0.005472   0.020857      0.023889  0.994647  1.000000   
Y2016          0.005247   0.020109      0.023444  0.996081  0.995739   
Y2017          0.005006   0.021494      0.024254  0.995230  0.988048   
Y2018          0.005665   0.021314      0.024279  0.994872  0.988208   

                 Y2016     Y2017     Y2018  
Area Code     0.005247  0.005006  0.005665  
Item Code     0.020109  0.021494  0.021314  
Element Code  0.023444  0.024254  0.024279  
Y2014         0.996081  0.995230  0.994872  
Y2015         0.995739  0.988048  0.988208  
Y2016         1.000000  0.992785  0.992757  
Y2017         0.992785  1.000000  0.998103  
Y2018         0.992757  0.998103  1.000000

Year 2014 have the highest correlation with 'Element Code ' with the correlation of 0.024457.

# 5. What year has the highest sum of Import Quantity?

In [10]:
Quantity = data[data['Element'] == 'Import Quantity']
Quantity[['Y2014','Y2015','Y2016','Y2017','Y2018']].sum()

Y2014    274144.48
Y2015    267018.46
Y2016    286582.78
Y2017    294559.09
Y2018    287997.09
dtype: float64

Year 2017 had the highest sum of Import Quantity with 294559.09.

# 6. What is the total number of the sum of Production in 2014?

In [11]:
production = data[data['Element'] == 'Production']
production['Y2014'].sum()

1931287.75

The sum of production for the year 2014 is 1931287.75

# 7.Which of these elements had the highest sum in 2018?

In [12]:
data.groupby('Element')['Y2018'].sum()

Element
Domestic supply quantity                  2161192.10
Export Quantity                            181594.80
Fat supply quantity (g/capita/day)          10258.69
Feed                                       233489.68
Food                                      1303841.28
Food supply (kcal/capita/day)              455261.00
Food supply quantity (kg/capita/yr)         49056.85
Import Quantity                            287997.09
Losses                                     163902.00
Other uses (non-food)                       91300.97
Processing                                 308429.00
Production                                2075072.89
Protein supply quantity (g/capita/day)      11833.56
Residuals                                   34864.00
Seed                                        25263.14
Stock Variation                             20577.91
Total Population - Both sexes             1140605.00
Tourist consumption                            90.00
Name: Y2018, dtype: float64

Amongst the Elements, Domestic supply quantity had the highest sum in 2018 with a total of 2161192.10

# 8. Which of these elements had the 3rd lowest sum in 2018?

In [13]:
data.groupby('Element')['Y2018'].sum().sort_values()

Element
Tourist consumption                            90.00
Fat supply quantity (g/capita/day)          10258.69
Protein supply quantity (g/capita/day)      11833.56
Stock Variation                             20577.91
Seed                                        25263.14
Residuals                                   34864.00
Food supply quantity (kg/capita/yr)         49056.85
Other uses (non-food)                       91300.97
Losses                                     163902.00
Export Quantity                            181594.80
Feed                                       233489.68
Import Quantity                            287997.09
Processing                                 308429.00
Food supply (kcal/capita/day)              455261.00
Total Population - Both sexes             1140605.00
Food                                      1303841.28
Production                                2075072.89
Domestic supply quantity                  2161192.10
Name: Y2018, dtype: float64

The Element with the third lowest sum is  'Protein supply quantity (g/capita/day)' with 11833.56

# 9.What is the total Import Quantity in Algeria in 2018?

In [14]:
element= data[data['Element']== 'Import Quantity']
element.groupby('Area')['Y2018'].sum().head()

Area
Algeria         36238.29
Angola           7769.67
Benin            7008.24
Botswana         1908.42
Burkina Faso     2605.64
Name: Y2018, dtype: float64

The total import quantity in Algeria in 2018 is 36238.29

# 10. What is the total number of unique countries in the dataset?

In [15]:
data.Area.nunique()

49

The total number of unique countries in the dataset is 49.